enterprise/backend/src/metabase_enterprise/metabot_v3/context.clj

```diff
@@ -130,6 +131,24 @@
       (assoc context :user_is_viewing enhanced-viewing))
     context))
 
+(defn- annotate-transform-source-types
+  "Annotate transforms in context with source types if not already present (e.g. for draft transforms not yet saved)"
+  [context]
+  (if-let [user-viewing (get context :user_is_viewing)]
+    (let [annotated-viewing
+          (mapv (fn [item]
+                  (try
+                    (if (and (= (:type item) "transform")
+                             (not (:source_type item)))
+                      (assoc item :source_type (transform/compute-transform-source-type (:source item)))
+                      item)
+                    (catch Exception e
+                      (log/error e "Error annotating transform source type for metabot context")
+                      item)))
```

Comment:
Kinda relates to [this question](https://github.com/metabase/ai-service/pull/441#discussion_r2481576884). If we catch the exception it sounds like we might still return `null`?

Reply:
Discussed on slack